x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/view/details/components/activity_log_date_range_picker/index.tsx

```diff
@@ -0,0 +1,135 @@
+import styled from 'styled-components';
+import moment from 'moment';
+import { EuiFlexGroup, EuiFlexItem, EuiDatePicker, EuiDatePickerRange } from '@elastic/eui';
+
+import * as i18 from '../../../translations';
+import { useEndpointSelector } from '../../../hooks';
+import { getActivityLogDataPaging } from '../../../../store/selectors';
+
+const DatePickerWrapper = styled.div`
+  width: ${(props) => props.theme.eui.fractions.single.percentage};
+  background: white;
+`;
+const StickyFlexItem = styled(EuiFlexItem)`
+  position: sticky;
+  top: ${(props) => props.theme.eui.euiSizeM};
+  z-index: 1;
+`;
+
+export const DateRangePicker = memo(() => {
+  const dispatch = useDispatch();
+  const { page, pageSize, startDate, endDate } = useEndpointSelector(getActivityLogDataPaging);
+
+  const isInvalidDateRange = useMemo(
+    () => (startDate && endDate ? moment(startDate) > moment(endDate) : false),
+    [startDate, endDate]
+  );
+
+  const onClear = useCallback(
+    ({ clearStart = false, clearEnd = false }: { clearStart?: boolean; clearEnd?: boolean }) => {
+      dispatch({
+        type: 'endpointDetailsActivityLogUpdatePaging',
+        payload: {
+          disabled: false,
+          page,
+          pageSize,
+          startDate: clearStart ? undefined : startDate,
+          endDate: clearEnd ? undefined : endDate,
+        },
+      });
+    },
+    [dispatch, endDate, startDate, page, pageSize]
+  );
+
+  const onChangeStartDate = useCallback(
+    (date) => {
+      if (isInvalidDateRange) {
+        return;
+      }
+      dispatch({
+        type: 'endpointDetailsActivityLogUpdatePaging',
+        payload: {
+          disabled: false,
+          page,
+          pageSize,
+          startDate: date ? date?.toISOString() : undefined,
+          endDate: endDate ? endDate : undefined,
+        },
+      });
+    },
+    [dispatch, isInvalidDateRange, endDate, page, pageSize]
+  );
+
+  const onChangeEndDate = useCallback(
+    (date) => {
+      if (isInvalidDateRange) {
+        return;
+      }
+      dispatch({
+        type: 'endpointDetailsActivityLogUpdatePaging',
+        payload: {
+          disabled: false,
+          page,
+          pageSize,
+          startDate: startDate ? startDate : undefined,
+          endDate: date ? date.toISOString() : undefined,
+        },
+      });
+    },
+    [dispatch, isInvalidDateRange, startDate, page, pageSize]
+  );
+
+  return (
+    <StickyFlexItem grow={false}>
+      <EuiFlexGroup justifyContent="flexEnd" responsive>
+        <DatePickerWrapper>
+          <EuiFlexItem>
+            <EuiDatePickerRange
+              data-test-subj="activityLogDateRangePicker"
+              startDateControl={
+                <EuiDatePicker
+                  aria-label="Start date"
+                  endDate={endDate ? moment(endDate) : undefined}
+                  isInvalid={isInvalidDateRange}
+                  maxDate={moment(startDate) || moment()}
```

Comment:
Why the maxDate for this can be `startDate`?

Reply:
Good catch. Fixed it in fe043611f64d025e4decb55acae48998328c8c6d